### PR TITLE
Add DASH manifest UTCTiming http-head scheme support

### PR DESCRIPTION
**Pull request by ghost**
Extend ExoPlayer DashMediaSource to support UTC timing scheme urn:mpegdashutc:http-head:2012, urn:mpegdashutc:http-head:2014

Related articles:
# https://www.google.com/patents/US20140195651
# http://vm2.dashif.org/dash.js/docs/jsdocs/TimeSyncController.js.html

### CODE REVIEW

**Comment by ghost** [comment]
@ojw28, please, comment this pull request. We need this functionality for more precise time sync.

File: library/dash/src/main/java/com/google/android/exoplayer2/source/dash/DashMediaSource.java [hunk 1]
```diff
@@ -585,6 +586,16 @@ private void resolveUtcTimingElement(UtcTimingElement timingElement) {
     } else if (Util.areEqual(scheme, "urn:mpeg:dash:utc:http-xsdate:2014")
         || Util.areEqual(scheme, "urn:mpeg:dash:utc:http-xsdate:2012")) {
       resolveUtcTimingElementHttp(timingElement, new XsDateTimeParser());
+    } else if (Util.areEqual(scheme, "urn:mpeg:dash:utc:http-head:2012")
+        || Util.areEqual(scheme, "urn:mpeg:dash:utc:http-head:2014")) {
+      // According to US 9426196 B2 and ISO 23009-1, url could be a white-space
```
Comment (ojw28) [inline library/dash/src/main/java/com/google/android/exoplayer2/source/dash/DashMediaSource.java:591]: I can't find any references to http-head in ISO 23009-1:2014. Could you clarify exactly where this is defined? Please can we also reference specifications only (and not patents)? Thanks!

File: library/dash/src/main/java/com/google/android/exoplayer2/source/dash/DashMediaSource.java [hunk 2]
```diff
@@ -585,6 +586,16 @@ private void resolveUtcTimingElement(UtcTimingElement timingElement) {
...
+      // separated list of URLs.
+      // As workaround for current architecture, just handle first one.
+      // TODO: find solution to support multiple urls, covered by US 9426196 B2 & ISO 23009-1
```
Comment (ojw28) [inline library/dash/src/main/java/com/google/android/exoplayer2/source/dash/DashMediaSource.java:594]: Not really sure what this means, although perhaps once I find the right spec all will become clear :).

File: library/dash/src/main/java/com/google/android/exoplayer2/source/dash/DashMediaSource.java [hunk 3]
```diff
@@ -983,4 +994,13 @@ public Long parse(Uri uri, InputStream inputStream) throws IOException {
 
   }
 
+  private static final class HttpHeadTimeParser implements ParsingLoadable.Parser<Long> {
+
+    @Override
+    public Long parse(Uri uri, InputStream inputStream) throws IOException {
+      return new URL(uri.toString()).openConnection().getDate();
```
Comment (ojw28) [inline library/dash/src/main/java/com/google/android/exoplayer2/source/dash/DashMediaSource.java:1001]: We could do with finding a way for this request to go through `dataSource`. If an application has opted to use `OkHttp` then we don't really want this one connection going through a different stack.

**Comment by ojw28** [comment]
Please see a couple of comments inline. At a high level:

* Yes, we should support this.
* The change needs to be going into the `dev-v2` branch, not `dev-v2-r2.6.1`.

Thanks!

**Comment by ojw28** [comment]
Closing because `dev-v2-r2.6.1` is being removed (and this is not possible whilst there's an open pull request). If you plan on addressing the comments above, please resend the amended request to the `dev-v2` branch, which is the correct branch for all requests. Thanks!

**Pull request closed without merging**